### PR TITLE
Orchestrator

### PR DESCRIPTION
**Pull request by theAbishek718**
PR for orchestrator so that it gets reviewed and ok'd. We will be tuning values later due to time constraints.

### CODE REVIEW

File: src/main/java/frc/robot/Orchestrator.java
Comment (admlvntv) [inline src/main/java/frc/robot/Orchestrator.java:1]: add some javadoc comments to all the methods

File: src/main/java/frc/robot/Orchestrator.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        indexer.setVolts(4),
+        arm.toSetpoint(ArmConstants.HORIZONTAL_OFFSET), // TODO: Make setpoint for pickup position.
+        Commands.waitUntil(() -> arm.atSetpoint() && pixy2.seesNote()).withTimeout(2),
+        intake.intake().until(() -> robotState.hasNote));
+
+  }
+
+  public Command expelFullRobot() {
+    return Commands.parallel(
+        shooter.manualShoot(-5), indexer.setVolts(-5.0), intake.release());
```
Comment (admlvntv) [inline src/main/java/frc/robot/Orchestrator.java:118]: do expelShindex and expelIntake in parallel rather than this

**Review by admlvntv — CHANGES_REQUESTED** [review]
some more fixes

File: src/main/java/frc/robot/Orchestrator.java (hunk 1 above)
Comment (theAbishek718) [reply src/main/java/frc/robot/Orchestrator.java]: Fixed

File: src/main/java/frc/robot/Orchestrator.java
Comment (theAbishek718) [reply src/main/java/frc/robot/Orchestrator.java:1]: Added

**Review by Aradig26 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Orchestrator.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+  /**
+   * Shoots with a velocity setpoint of 9999 rad per seconds, this needs to be tuned. Turns off the
+   * indexer and the shooter when the limit switch is not pressed.
+   *
+   * @return The command to shoot the note.
+   */
+  public Command shootBasic() {
+    return Commands.sequence(
+            Commands.parallel(shooter.manualShoot(12)),
```
[the remark this answers is not in the document]
Comment (theAbishek718) [reply src/main/java/frc/robot/Orchestrator.java:108]: Yea I'll change that, there was originally something else there but I had moved it and forgot to take out the parallel part. Thanks for spotting.

File: src/main/java/frc/robot/Orchestrator.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+            Commands.waitUntil(shooter::ShooterSpeedIsReady).withTimeout(2),
+            indexer.setVolts(1),
+            Commands.waitUntil(() -> !indexer.getLimitSwitchPressed()).withTimeout(2),
+            Commands.parallel(indexer.setVolts(0), shooter.manualShoot(0)))
+        .onlyIf(indexer::getLimitSwitchPressed);
```
[the remark this answers is not in the document]
Comment (Aradig26) [reply src/main/java/frc/robot/Orchestrator.java:113]: nvm

**Comment by Aradig26** [comment]
Outdated. The tested and up-to-date version is in origin/everything-bagel

**Pull request closed without merging**